python/pyarrow/array.pxi

```diff
@@ -118,11 +118,12 @@ def _handle_arrow_array_protocol(obj, type, mask, size):
     return res
 
 
-def _handle_run_end_encoded_arrays(obj, type):
+cdef _handle_run_end_encoded_arrays(object sequence, object mask, object size,
+                                    DataType type, CMemoryPool* pool,
+                                    c_bool from_pandas):
     from pyarrow.compute import run_end_encode
-    ree_arr = run_end_encode(obj)
-    return RunEndEncodedArray.from_arrays(
-        ree_arr.run_ends.to_pylist(), ree_arr.values.to_pylist(), type)
+    arr = _sequence_to_array(sequence, mask, size, type.value_type, pool, from_pandas)
```

Comment:
Here, we now use the more generic `_sequence_to_array`, also when the input is a ndarray. So maybe something like:

```
if isinstance(sequence, np,ndarray):
     arr = _ndarray_to_array(values, mask, type, c_from_pandas, safe, pool)
else:
    arr = _sequence_to_array(sequence, mask, size, type.value_type, pool, from_pandas)
```

?

Reply:
Thinking that the code for each of the two separate cases (sequence and array-like) is now minimal and maybe we do not need a helper function anymore?

Reply:
That seems fine to me! Could we move the import to the beginning of the file or is there a benefit to keeping it within scope of this code?

Reply:
No, no benefit, I think. WIll move the import to the beginning of the file.